File: VB/B3013/MyTreeList.vb
Imports DevExpress.XtraTreeList
Imports DevExpress.XtraTreeList.ViewInfo
Imports System.Drawing
Imports DevExpress.XtraTreeList.Nodes
Imports DevExpress.XtraTreeList.Painter

Namespace DXSample
    Public Class MyTreeList
        Inherits TreeList

        Public Sub New()
            MyBase.New()
        End Sub

        Protected Overrides Function CreateViewInfo() As TreeListViewInfo
            Return New MyTreeListViewInfo(Me)
        End Function
    End Class

    Public Class MyTreeListViewInfo
        Inherits TreeListViewInfo

        Public Sub New(ByVal treeList As TreeList)
            MyBase.New(treeList)
        End Sub

        Protected Overrides Sub CalcSelectImageBounds(ByVal rInfo As RowInfo, ByVal indentBounds As Rectangle)
            MyBase.CalcSelectImageBounds(rInfo, indentBounds)
            If -1 = rInfo.SelectImageIndex Then
                rInfo.SelectImageBounds = Rectangle.Empty
            End If
        End Sub
        Protected Overrides Sub CalcStateImageBounds(ByVal rInfo As RowInfo, ByVal indentBounds As Rectangle)
            MyBase.CalcStateImageBounds(rInfo, indentBounds)
            If Size.Empty <> RC.SelectImageSize AndAlso -1 = rInfo.StateImageIndex Then
                rInfo.StateImageBounds = Rectangle.Empty
            End If
        End Sub
    End Class
End Namespace

File: CS/B3013/MyTreeList.cs
using DevExpress.XtraTreeList;
using DevExpress.XtraTreeList.ViewInfo;
using System.Drawing;
using DevExpress.XtraTreeList.Nodes;
using DevExpress.XtraTreeList.Painter;

namespace DXSample {
    public class MyTreeList :TreeList {
        public MyTreeList() : base() { }

        protected override TreeListViewInfo CreateViewInfo() {
            return new MyTreeListViewInfo(this);  
        }
    }

    public class MyTreeListViewInfo :TreeListViewInfo {
        public MyTreeListViewInfo(TreeList treeList) : base(treeList) { }

        protected override void CalcSelectImageBounds(RowInfo rInfo, Rectangle indentBounds)
        {
            base.CalcSelectImageBounds(rInfo, indentBounds);
            if (-1 == rInfo.SelectImageIndex) rInfo.SelectImageBounds = Rectangle.Empty;
        }
        protected override void CalcStateImageBounds(RowInfo rInfo, Rectangle indentBounds)
        {
            base.CalcStateImageBounds(rInfo, indentBounds);
            if (Size.Empty != RC.SelectImageSize && -1 == rInfo.StateImageIndex)
                rInfo.StateImageBounds = Rectangle.Empty;
        }
    }
}